composer.json

```diff
@@ -36,12 +36,14 @@
         "psr/log": "^1.0.0",
         "psr/simple-cache": "^1.0.0",
         "zendframework/zend-diactoros": "^1.4.0",
-        "paragonie/random_compat": "^2.0|9.99.99"
+        "paragonie/random_compat": "^1.4|9.99.99",
+        "paragonie/csp-builder": "^1.4"
```

Comment:
It could be `"paragonie/random_compat": "^1.4|^2.0|9.99.99"`

- 1.4 when on PHP 5.6 and `paragonie/csp-builder` (unfortunately there is no tagged release with relaxed dependency https://github.com/paragonie/csp-builder/compare/v1.4.0...v1.x)
- 2.0 when  on PHP 5.6 without `paragonie/csp-builder`
- 9.99 when on PHP 7

And `paragonie/csp-builder` should be moved to `require-dev` and updated to `^1.4|^2.3` as it is optional.

Reply:
Thanks for the feedback! I've struggled with how I should do this. Great suggestions!